controller/src/beerocks/master/wfa_ca.cpp

```diff
@@ -22,12 +111,54 @@ void wfa_ca::handle_wfa_ca_message(
         return;
     }
 
-    auto command_tuple = request->command(0);
-    if (!std::get<0>(command_tuple)) {
-        LOG(ERROR) << "getting command entry has failed!";
-        return;
-    }
-    const char *command_ptr = (char *)(&std::get<1>(command_tuple));
+    const char *command_ptr = request->command();
 
     LOG(DEBUG) << "Received ACTION_BML_WFA_CA_CONTROLLER_REQUEST, command=" << command_ptr;
+
+    std::string command(command_ptr);
+    auto cmd_tokens_vec = string_utils::str_split(command_ptr, ',');
+
+    if (cmd_tokens_vec.size() < 1) {
+        LOG(ERROR) << "Invalid WFA-CA command";
+        reply_invalid_error(sd, cmdu_tx, eWfaCaStatus::INVALID, "empty command");
+    }
+
+    auto command_type     = wfa_ca_command_from_string(cmd_tokens_vec[0]);
+    auto command_type_str = cmd_tokens_vec[0];
```

Comment:
It would be more logical to swap:

```
auto command_type_str = cmd_tokens_vec[0];
auto command_type = wfa_ca_command_from_string(command_type_str);
```

(nitpick, obviously)

Reply:
will change it